benches/encryption_bench.rs

```diff
@@ -0,0 +1,156 @@
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use tokio::runtime::Runtime;
+use rustc_hash::FxHashMap;
+use cripta::{
+    app::AppState,
+    config,
+    core::crypto::custodian::Custodian,
+    types::{
+        core::{DecryptedData, DecryptedDataGroup, Identifier},
+        method::EncryptionType,
+    },
+};
+
+const SINGLE_BENCH_ITERATION: u32 = 10;
+const BATCH_BENCH_ITERATION: u32 = 10;
+
+criterion_main!(benches);
+criterion_group!(benches, criterion_data_encryption_decryption, criterion_batch_data_encryption_decryption);
+
+pub fn criterion_data_encryption_decryption(c: &mut Criterion) {
+    let rt = Runtime::new().unwrap();
+    let custodian = Custodian::new(Some(("key".to_string(),"value".to_string())));
+    let config = config::Config::with_config_path(config::Environment::which(), None);
+    let state = rt.block_on(async {
+         let state = AppState::from_config(config).await;
+         return state;
+    });
+    let identifier = Identifier::User(String::from("user_12345"));
+
+    {
+        let mut group = c.benchmark_group("data-encryption-single");
```

Comment:
Can you create the key `user_12345` in the DB before running the bench

Reply:
Yes, I created a new DataKey using the following api and ran the benchmark, will update the description
```
curl --location 'localhost:5000/key/create' \
--header 'Content-type: application/json' \
--data '{
  "data_identifier": "User",
  "key_identifier": "user_12345"
}'
```

Reply:
<img width="1272" alt="Screenshot 2024-11-04 at 11 45 58 AM" src="https://github.com/user-attachments/assets/83035280-1332-4716-af74-67fc77ad506b">

Reply:
@dracarys18 added the logic to create user at the time of running benchmark. Can you review it again?